src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+ * Called when a particular card has to be selected for removing from a deck
+ * Frees the menory allocated to he temporary selected drop card input parameter
+ * Note: Function  must be called only if the card to be selected is VALID
+ * 
+ * @param pp_head :Pointer which points to the address of head of card list in the Deck
+ * @param card_to_be_matched :Card which has to be matched from the deck
+ * 
+ * @return Card_t* Card that has been matched with the given card value
+ *         NULL - Returns NULL if the given card does not match with any of the cards in the deck
+ */
+Card_t *pick_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_matched)
+{
+    Deck_t *temp = *pp_head;
+    while (temp->next != NULL)
+    {
+        if (temp->card.color == card_to_be_matched->color && temp->card.name == card_to_be_matched->name)
+        {
+            return &temp->card;
+        }
+        temp = temp->next;
+    }
+    free(card_to_be_matched); //Check if it can be freed here
+    return NULL;
+}
+
+/**
+ * @brief Displays the card assigned to the human player on the console during Request card/
+ * Draw1 and Draw2 scenarios
+ * 
+ * @param assigned_card Card that has been added newly to the player's deck
+ */
+void show_cards_assigned(Card_t assigned_card)
+{
+    printf("The card from the draw pile is %d%d \n", assigned_card.color, assigned_card.name);
+    display_player_deck(human_player);
+    return;
+}
+
+/**
+ * @brief Checks if the card dropped by human player during his turn is valid
+ *
+ * @param previous_card - Card that was dropped previously in the pile
+ * @param current_card - Current card dropped by the human player
+ * 
+ * @return- True if the card is invaid
+ *          False if the card is valid
+ */
+bool check_is_valid_turn(Card_t *previous_card, Card_t *current_card)
+{
+    bool is_valid = false;
+    if (g_discard_pile == NULL)
+    {
+    }
+    if ((previous_card->color == current_card->color) || (previous_card->name == current_card->name))
+    {
+        is_valid = true;
+    }
+    return is_valid;
+}
+
+/**
+ * @brief Determines next player based on the previous card dropped on the discard pile
+ *
+ * @param previous_card - Last dropped card on the pile
+ * @param current_player - Indicates the current player who dropped the card  
+ */
+
+PlayerType_e determine_next_player(struct CARD *previous_card, PlayerType_e current_player)
+{
+    PlayerType_e next_player;
+    if (current_player == 0 && !is_action_card)
+    {
+        next_player = COMPUTER;
+        //call function to execute the computer's turn
+    }
+    else if (current_player == 1 && !is_action_card)
+    {
+        next_player = HUMAN;
+        handle_human_turn();
+    }
+    else
+    {
+        /* Action card functionality in Release 2 */
+    }
+    return next_player;
+}
+
+/**
+ * @brief This function handles the functionality to support human player's turn
+ * 
+ */
+error_e handle_human_turn()
```

Comment:
Same comments as above: If a function is not meant to take any parameters, specify that by using void in the parameter list.

Reply:
Changes done in new pull req.